### PR TITLE
Javascript Testing & Build for CI

### PR DESCRIPTION
**Pull request by larkmullins**

### CODE REVIEW

File: ui/config/karma.conf.js [hunk 1]
```diff
@@ -7,7 +7,8 @@ module.exports = function (config) {
     frameworks: ['jasmine'],
 
     files: [
-      {pattern: './config/karma-test-shim.js', watched: false}
+      {pattern: './config/karma-test-shim.js', watched: false},
+      'tests/*.test.js'
```
Comment (kyleferguson) [inline ui/config/karma.conf.js]: I think a common convention for JS right now is `*.spec.js`

File: wercker.yml [hunk 2]
```diff
@@ -28,6 +30,27 @@ ci:
...
+# build and test UI
+ci-ui:
+  box: node
+  steps:
+    - script:
+        name: install and build
+        code: |
+          cd ui
+          npm install
+          npm build
```
Comment (jlaswell) [inline wercker.yml]: Why run `npm build`?

File: wercker.yml [hunk 3]
```diff
@@ -28,6 +30,27 @@ ci:
...
+    - script:
+        name: run tests
+        code: |
+          cd ui
+          npm test
+    - script:
+        name: clean up
+        code: |
+          cd ui
+          rm -rf ./node_modules
```
Comment (kyleferguson) [inline wercker.yml]: We don't do anything with the CI containers after they're done so we don't need to worry about removing node_modules or cleanup etc.

**Review by kyleferguson — CHANGES_REQUESTED** [review]
Couple minor changes, but otherwise looks great!

File: wercker.yml (hunk 2 above)
Comment (kyleferguson) [reply wercker.yml]: I don't mind doing that since the dev compile process is quite different, will verify the code is able to be compiled for production.

Comment (jlaswell) [reply wercker.yml]: Ah, if we want to run the build command we need to run `npm run build`. `npm build` is an internal command that npm uses during install and other stuff

Comment (kyleferguson) [reply wercker.yml]: Good catch, missed that

**Review by kyleferguson — APPROVED** [review]

**Pull request merged**